internal/server/middleware.go

```diff
@@ -162,15 +162,15 @@ func requireAccessKey(c *gin.Context, db *data.Transaction, srv *Server) (access
 	accessKey, err := data.ValidateRequestAccessKey(db, bearer)
 	if err != nil {
 		if errors.Is(err, data.ErrAccessKeyExpired) {
-			return u, err
+			return u, AuthenticationError{Message: "access key has expired"}
```

Comment:
I think previously this would have returned a `500` response, because I don't see `data.ErrAccessKeyExpired` being handled by `sendAPIError`. Not it should be a 401 with this message to help the user understand why it's failing.